salt-master/README.md

```diff
@@ -0,0 +1,46 @@
+Build salt-master image
+=======================
+
+```
+docker build -t moonshot/salt-master metalk8s/salt-master
+```
+
+Run salt-master container
+=========================
+
+* Create persistent data directory
+```
+mkdir /path/to/persistent/data/salt-master
+```
+
+* Normal mode
+```
+docker run --name salt-master \
+    --rm --detach \
+    -v /path/to/persistent/data/salt-master:/etc/salt \
+    -p 4505:4505 \
+    -p 4506:4506 \
+    moonshot/salt-master
+```
+
+* Debug mode
+```
+docker run --name salt-master \
+    --rm --tty \
+    -v /path/to/persistent/data/salt-master:/etc/salt \
+    -p 4505:4505 \
+    -p 4506:4506 \
+    --env SALT_LOG_LEVEL=deubg \
+    moonshot/salt-master
+```
+
+Run salt commands in container
```

Comment:
How do we run them from outside the container?

Reply:
So for this, I imagine that a custom package would wrap "salt" command (and possibly even conflict with salt-master package, this way no mistake can be done) and this "wrapper" automatically calls for the local container with the salt-master.

Possibly configure by default a condition like:
```
if which crictl &>/dev/null ; then
    <use crictl>
elif which docker &>/dev/null ; then
    <use docker>
fi
```

Reply:
I disagree. Would rather bind-mount the directory containing the UDSs for master communication from the host into the container. Same, by the way, for cache, configuration,...

Reply:
Don't get it... Is is just a matter of wording, or something else ?

Reply:
It's not just wording, we'll want to have various things used by the master to be accessible from the outside, or shared with other containers in a Pod, e.g. `salt-api` will need access to the `salt-master` socket, and may run as another container in the same Pod.

Also, we'll want `/var/run/salt`, `/var/cache/salt` to survive master (container) restarts.

Also, ideally all our containers can run with a read-only root-fs (`--read-only` flag of Docker), so we may need to bind in a tmpfs for `/tmp`, `/var/tmp`,...

Reply:
We can tackle this at the 'runtime' stage.